File: src/main/java/com/example/studentmanagementsystem/mappers/StudentMapper.java
package com.example.studentmanagementsystem.mappers;

import com.example.studentmanagementsystem.entity.Student;
import java.util.List;
import org.apache.ibatis.annotations.Param;
import org.springframework.stereotype.Repository;

/** @author Hingbong */

@Repository
public interface StudentMapper {

  /**
   * add new student
   *
   * @param student which will be added
   * @return rows affected
   */
  Integer addNewStudent(Student student);

  /**
   * list all students
   *
   * @param profession profession
   * @param stuName stuName
   * @return all students
   */
  List<Student> findAllStudent(
      @Param("stuName") String stuName, @Param("profession") Integer profession);

  /**
   * delete student by id
   *
   * @param stuId student id
   * @return rows affected
   */
  Integer deleteStudentById(Integer stuId);

  /**
   * modify student info
   *
   * @param student student info
   * @return rows affected
   */
  Integer modifyStudent(Student student);

  /**
   * find student by id
   *
   * @param stuId student id
   * @return student
   */
  Student findStudentById(Integer stuId);

  /**
   * get all student id
   *
   * @return all student id
   */
  List<Integer> findAllStuId();

  /**
   * when get score set the total score
   *
   * @param stuId student
   * @param totalScore total score
   * @return rows affected
   */
  Integer setTotalScoreById(@Param("stuId") Integer stuId, @Param("totalScore") Integer totalScore);
}

File: src/main/kotlin/com/example/studentmanagementsystem/mappers/StudentMapper.kt
package com.example.studentmanagementsystem.mappers

import com.example.studentmanagementsystem.entity.Student
import org.apache.ibatis.annotations.Param
import org.springframework.stereotype.Repository

/** @author Hingbong
 */

@Repository
interface StudentMapper {

    /**
     * add new student
     *
     * @param student which will be added
     * @return rows affected
     */
    fun addNewStudent(student: Student): Int

    /**
     * list all students
     *
     * @param profession profession
     * @param stuName stuName
     * @return all students
     */
    fun findAllStudent(
            @Param("stuName") stuName: String?, @Param("profession") profession: Int?): List<Student>

    /**
     * delete student by id
     *
     * @param stuId student id
     * @return rows affected
     */
    fun deleteStudentById(stuId: Int): Int

    /**
     * modify student info
     *
     * @param student student info
     * @return rows affected
     */
    fun modifyStudent(student: Student): Int

    /**
     * find student by id
     *
     * @param stuId student id
     * @return student
     */
    fun findStudentById(stuId: Int): Student

    /**
     * get all student id
     *
     * @return all student id
     */
    fun findAllStuId(): List<Int?>

    /**
     * when get score set the total score
     *
     * @param stuId student
     * @param totalScore total score
     * @return rows affected
     */
    fun setTotalScoreById(@Param("stuId") stuId: Int, @Param("totalScore") totalScore: Int): Int
}
